fusillade/resource.py

```diff
@@ -0,0 +1,225 @@
+            raise FusilladeHTTPException('Must provide owner policy.')
+        else:
+            if owner_policy:
+                owner_policy = new_node.format_policy(owner_policy)
+                verify_iam_policy(owner_policy)
+            elif getattr(cls, '_default_policy_path'):
+                with open(cls._default_policy_path, 'r') as fp:
+                    owner_policy = json.load(fp)
+            ops.extend(new_node.create_policy('Owner',
+                                              owner_policy,
+                                              parent_path='#policy_node',
+                                              run=False,
+                                              type=new_node.object_type,
+                                              name=f"{new_node.name}:Owner"))
+
+        try:
+            new_node.cd.batch_write(ops)
+        except cd_client.exceptions.BatchWriteException as ex:
+            if 'LinkNameAlreadyInUseException' in ex.response['Error']['Message']:
+                raise FusilladeHTTPException(
+                    status=409, title="Conflict", detail=f"The {cls.object_type} named {name} already exists. "
+                    f"{cls.object_type} was not modified.")
+            else:
+                raise FusilladeHTTPException(ex)
+        else:
+            new_node.cd.get_object_information(new_node.object_ref,
+                                               ConsistencyLevel=ConsistencyLevel.SERIALIZABLE.name)
+            logger.info(dict(message=f"{new_node.object_type} created by {_creator}",
+                             object=dict(type=new_node.object_type, path_name=new_node._path_name)))
+            logger.info(dict(message="Policy updated",
+                             object=dict(
+                                 type=new_node.object_type,
+                                 path_name=new_node._path_name
+                             ),
+                             policy=dict(
+                                 link_name='Owner',
+                                 policy_type='IAMPolicy')
+                             ))
+            return new_node
+
+    # TODO: Add function to modify actions, this will need to modify all access policies with this actions
+    @property
+    def actions(self):
+        if not self._actions:
+            self._get_attributes(self._attributes)
+        if isinstance(self._actions, str):
+            self._actions = self._actions.split(' ')
+        return self._actions
+
+    def add_actions(self):
+        set(self._actions)
+
+    def check_actions(self, policy: dict):
+        policy_actions = set()
+        for s in policy['Statement']:
+            policy_actions.update(s['Action'])
+        if not policy_actions.issubset(set(self.actions)):
+            raise FusilladeBadRequestException(detail="Invalid actions in policy.")
+
+    @staticmethod
+    def hash_name(name):
+        return name
```

Comment:
Is this method intended to just return the parameter?

Reply:
this is overriding the hash_name function in CloudNode. I'll add documentation to clarify